pkg/controller/controller.go

```diff
@@ -561,6 +563,30 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 		return nil, controller.ProvisioningInBackground, capErr
 	}
 
+	if options.PVC.Spec.DataSource != nil && (rc.clone || rc.snapshot) {
+		contentSource := rep.GetVolume().ContentSource
+		if contentSource == nil {
+			sourceErr := fmt.Errorf("volume content source missing")
+			delReq := &csi.DeleteVolumeRequest{
+				VolumeId: rep.GetVolume().GetVolumeId(),
+			}
+			delReq.Secrets = provisionerCredentials
+			ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
+			defer cancel()
+			for i := 0; i < deleteVolumeRetryCount; i++ {
```

Comment:
@xing-yang pointed out that the above Deletion method should also retry too. Can you put this into a common method and then have it called in both places?

Reply:
@msau42 I will do it